lib/barbeque_client/executor.rb

```diff
@@ -8,9 +8,15 @@ class Executor
     # @param [String] queue_name - barbeque's job_queues.name
     def initialize(job:, message:, message_id:, queue_name:)
       @job        = job
-      @message    = JSON.load(message)
       @message_id = message_id
       @queue_name = queue_name
+
+      parsed_message = JSON.load(message)
+      if parsed_message.is_a?(Array)
```

Comment:
We need some comments that describe what kind of case is expected for following two cases.